### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by adambirtles**
Just a bunch of small improvements:

* Made code style nicer and more consistent
    * Limited line length to 80 columns
    * Made pointers all `type *name`
    * Fixed indentation inconsistency (my fault, oops)
    * Various other bits and bobs
* Got rid of namespace polution (i.e. `using namespace std`)
* Changed some magic numbers to constants with meaningful names`
* Added a makefile and updated README instructions to match

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 *
```
Comment (LordAro) [inline .gitignore:1]: gitignore ignoring by default is unsual, why this way round?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+target := uoyweek
+source := uoyweek.cpp
+
+.DEFAULT_GOAL := $(target)
+
+.PHONY: run install clean
+
+$(target): $(source)
+	$(CXX) -Wall -Wextra -pedantic $< -o $@
```
Comment (LordAro) [inline Makefile:11]: factor out into CPPFLAGS variable?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	chmod +x $@
```
Comment (LordAro) [inline Makefile:12]: gcc/clang will do this for you

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+run: $(target)
+	./$(target) $(ARGS)
+
+install: $(target)
+	cp $^ $(PREFIX)
+
+clean:
+	$(RM) $(target)
```
Comment (LordAro) [inline Makefile:21]: RM looks undefined, `rm -f` usually suffices (-f for cleaning when the target doesn't exist)

File: README.md [hunk 5]
```diff
@@ -1,20 +1,30 @@
 # UoYWeek
-UoYWeek gives the current date string formatted as used by the [University of York](https://www.york.ac.uk/). For example, `spr/3/wed` would refer to Wednesday on the 3rd week of Spring term.
+
+UoYWeek gives the current date string formatted as used by the [University of
+York](https://www.york.ac.uk/). For example, `spr/3/wed` would refer to
```
Comment (LordAro) [inline README.md:4]: Leaving aside the issue of wrapping at 80chars, i wouldn't do it in the middle of a link

File: uoyweek.cpp [hunk 6]
```diff
@@ -19,7 +34,8 @@ class Term {
         code = c;
         startTS = s;
         endTS = e;
-        //Store actual value instead of reference to gmtime so as to avoid issues
+        /* Store actual value instead of reference to gmtime so as to avoid
+         * issues */
```
Comment (LordAro) [inline uoyweek.cpp:38]: something something 80 chars wrap. This is why I prefer 120 chars, or at least trying to match the line lengths

File: uoyweek.cpp [hunk 7]
```diff
@@ -30,47 +46,49 @@ class Term {
...
-    "sunday",
-    "monday",
-    "tuesday",
-    "wednesday",
-    "thursday",
-    "friday",
-    "saturday"
-};
+const int AUT_START = 1506297600l;
+const int AUT_END = 1512086400l;
```
Comment (LordAro) [inline uoyweek.cpp:68]: alignment

File: uoyweek.cpp [hunk 8]
```diff
@@ -30,47 +46,49 @@ class Term {
...
     return result;
 }
 
-int getWeek(time_t now, Term * currentTerm) {
+int getWeek(time_t now, Term *currentTerm) {
     time_t startNormalised = normaliseToMonday(currentTerm->startTS);
     time_t interval = now - startNormalised;
-    int weeks = interval/604800;
-    return weeks + 1;
+    return interval/SECS_PER_WEEK + 1;
```
Comment (LordAro) [inline uoyweek.cpp:88]: spacing

File: uoyweek.cpp [hunk 9]
```diff
@@ -83,19 +101,19 @@ Term * getTerm(time_t now) {
...
-	}
-    int length = strlen(term->code.c_str()) + 2 + 3 + (week>=10?2:1) + 1;
-    char* str = (char*) malloc(sizeof(char) * length);
+    string day = DAYS[gmtime(&now)->tm_wday].substr(0,3);
+    if (capitalize) {
+        code[0] = std::toupper(code[0]);
+        day[0] = std::toupper(day[0]);
+    }
+    int length = strlen(term->code.c_str()) + 2 + 3 + (week/10 + 1) + 1;
+    char *str = (char *)malloc(sizeof(char) * length);
```
Comment (LordAro) [inline uoyweek.cpp:112]: good lord no. not in C++. `new` at the very least, or even better, a stringstream

File: uoyweek.cpp [hunk 10]
```diff
@@ -83,19 +101,19 @@ Term * getTerm(time_t now) {
...
     snprintf(str, length, "%s/%d/%s", code.c_str(), week, day.c_str());
     return str;
 }
 
 int main(void) {
-    cout << getString(false) << endl;
+    std::cout << getString(false) << std::endl;
```
Comment (LordAro) [inline uoyweek.cpp:118]: Avoid endl unless you actually require a flush - '\n' suffices

File: uoyweek.cpp [hunk 11]
```diff
@@ -30,47 +46,49 @@ class Term {
...
-        //sprintf is fine for actual values but results of asctime, etc. should be put in string in memory first
-        sprintf(yearStr, "%s (%d-%d) %s %s", code.c_str(), ((&start)->tm_year+1900), ((&end)->tm_year+1900), start_string, end_string);
+        /* sprintf is fine for actual values but results of asctime, etc. should
+         * be put in string in memory first */
+        sprintf(yearStr, "%s (%d-%d) %s %s",
+                code.c_str(),
+                (&start)->tm_year + 1900,
+                (&end)->tm_year + 1900,
+                start_string,
+                end_string);
```
Comment (LordAro) [inline uoyweek.cpp:62]: Again, stringstreams instead of char arrays

File: uoyweek.cpp [hunk 12]
```diff
@@ -19,7 +34,8 @@ class Term {
         code = c;
         startTS = s;
         endTS = e;
-        //Store actual value instead of reference to gmtime so as to avoid issues
+        /* Store actual value instead of reference to gmtime so as to avoid
+         * issues */
         start = *gmtime(&s);
```
Comment (LordAro) [inline uoyweek.cpp:39]: dereferencing these looks a bit odd. Maybe a memory leak?

File: Makefile (hunk 4 above)
Comment (adambirtles) [reply Makefile:21]: From <https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html>:

> **RM**
> Command to remove a file; default ‘rm -f’.

File: uoyweek.cpp (hunk 12 above)
Comment (LukeMoll) [reply uoyweek.cpp:39]: `ctime` does some weird stuff with results. If you don't dereference immediately value is lost

File: uoyweek.cpp (hunk 10 above)
Comment (LukeMoll) [reply uoyweek.cpp:118]: TIL. Is this alright since it's the only thing that's getting printed or still unnecessary?

File: uoyweek.cpp (hunk 9 above)
Comment (LukeMoll) [reply uoyweek.cpp:112]: I'll look into stringstreams

File: uoyweek.cpp (hunk 7 above)
Comment (LukeMoll) [reply uoyweek.cpp:68]: This is prettified temp code, should be gone soon

File: uoyweek.cpp (hunk 8 above)
Comment (LukeMoll) [reply uoyweek.cpp:88]: `interval / SECS_PER_WEEK`?

Comment (adambirtles) [reply uoyweek.cpp:88]: `a/b + c` makes reading it easier a glance, IMO.

File: uoyweek.cpp (hunk 10 above)
Comment (LordAro) [reply uoyweek.cpp:118]: For this, it's indeed largely irrelevant, but can have a big impact - http://demin.ws/blog/english/2012/04/25/endl-vs-new-line/

File: Makefile (hunk 4 above)
Comment (LordAro) [reply Makefile:21]: huh, TIL

File: uoyweek.cpp (hunk 12 above)
Comment (LordAro) [reply uoyweek.cpp:39]: http://en.cppreference.com/w/cpp/chrono/c/gmtime
> Pointer to a static internal std::tm object on success

No memory leak then, but dereferencing is weird.

Also, below: `(&start)->tm_year`  wut.

File: uoyweek.cpp (hunk 8 above)
Comment (LordAro) [reply uoyweek.cpp:88]: Aye, use brackets if you're not comfortable "relying" on operator precedence

**Pull request merged**